src/tasks/netSdk/NetSdkTaskHelper.ts

```diff
@@ -0,0 +1,138 @@
+            withNamedArg('--configuration', configuration),
+            withNamedArg('-p:ContainerImageName', getValidImageName(folderName.name), { assignValue: true }),
+            withNamedArg('-p:ContainerImageTag', IMAGE_TAG, { assignValue: true })
+        )();
+
+        const quotedArgs = Shell.getShellOrDefault().quote(args);
+        return quotedArgs.join(' ');
+    }
+
+    public async getNetSdkRunCommand(context: DockerTaskExecutionContext): Promise<string> {
+        const client = await ext.runtimeManager.getClient();
+        const folderName = await quickPickWorkspaceFolder(context.actionContext, ERR_MSG_NO_WORKSPACE_FOLDER);
+
+        const command = await client.runContainer({
+            detached: true,
+            publishAllPorts: true,
+            name: getDefaultContainerName(folderName.name, IMAGE_TAG),
+            environmentVariables: {},
+            removeOnExit: true,
+            imageRef: getDefaultImageName(folderName.name, IMAGE_TAG),
+            labels: defaultVsCodeLabels,
+            mounts: await this.getRemoteDebuggerMount(),
+            customOptions: '--expose 8080',
+            entrypoint: '/bin/sh'
+        });
+
+        const quotedArgs = Shell.getShellOrDefault().quote(command.args);
+        const commandLine = [client.commandName, ...quotedArgs].join(' ');
+
+        return commandLine;
+    }
+
+    public async inferProjPath(context: IActionContext, folder: WorkspaceFolder): Promise<string> {
+        const noProjectFileErrMessage = l10n.t('No .csproj file could be found.');
+        const item = await quickPickProjectFileItem(context, undefined, folder, noProjectFileErrMessage);
+        return item.absoluteFilePath || '';
+    }
+
+    /**
+     * This method normalizes the Docker OS type to match the .NET Core SDK conventions.
+     * {@link https://learn.microsoft.com/en-us/dotnet/core/rid-catalog}
+     */
+    private async normalizeOsToRid(): Promise<'linux' | 'win'> {
+        const dockerOsType = await getDockerOSType();
+        return dockerOsType === 'windows' ? 'win' : 'linux';
+    }
+
+    /**
+     * This method normalizes the native architecture to match the .NET Core SDK conventions.
+     * {@link https://learn.microsoft.com/en-us/dotnet/core/rid-catalog}
+     */
+    private async normalizeArchitectureToRid(): Promise<RidCpuArchitecture> {
+        const architecture = os.arch();
+        switch (architecture) {
+            case 'x32':
+            case 'ia32':
+                return 'x86';
+            default:
+                return architecture;
+        }
+    }
+
+    /**
+     * This methods returns the mount for the remote debugger ONLY as the SDK built container will have
+     * everything it needs to run the app already inside.
+     */
+    private async getRemoteDebuggerMount(): Promise<RunContainerBindMount[] | undefined> {
+        const volumes: DockerContainerVolume[] = [];
+        const isLinux = await getDockerOSType() === 'linux';
```

Comment:
I'd suggest using isWindows as the primary condition you check against (similar to the behavior in NetCoreTaskHelper.ts), since there could theoretically be additional Docker OS types, such as darwin, but they will generally match Linux conventions. Treating Windows as the exceptional case is usually safer.